pkg/connector/connector.go

```diff
@@ -32,6 +32,7 @@ const (
 const (
 	ProvisionTypeAPI ProvisionType = iota
 	ProvisionTypeConfig
+	ProvisionTypeDLQ // used for provisioning DLQ connectors which are not persisted
```

Comment:
WDYT about having this as a connector type (i.e. we'd have `TypeSource`, `TypeDestination`, and `TypeDLQ`)? It feels like it's more accurate to say that a DLQ is a special type of connector than a special type of provisioning. And DLQs can be provisioned through configuration files and we'll be able to provision them through the API in the future as well.

Reply:
How many times you comment on something that I considered when working on my code, it's getting scary 😄

I thought about this, in fact, I think I even started with a connector type and changed it to a provision type. That's because the connector itself is still a normal destination connector, it has no idea that it's being used as a DLQ destination. What is different though is how it's provisioned/created - that happens only when the pipeline is actually started, it doesn't exist before like a normal destination connector. So in that sense, it's not provisioned by the config file or by the API, it's provisioned internally by Conduit when the pipeline starts.

Reply:
I know what you're saying, let me present a different view though. We introduced the field "provisioned by" to be able to know if a connector was created through the API or config file and to restrict further changes to the connector only through that mechanism. A _DLQ configuration_ works the same way - it's created, updated, deleted only through the API or config file. Notice I said _DLQ configuration_, that's not the same as the DLQ connector. When hitting the API endpoint for updating the DLQ, no connector is provisioned, it's only the configuration that changes. The DLQ connector provisioning is triggered only when the pipeline is started. We could call the type `ProvisionTypeInternal` if that makes it clearer.

The point is that the connector lifecycle is managed internally, it will be created and deleted based on the status of the pipeline. You can't change the _DLQ connector_ through the API or config file, regardless how the _DLQ configuration_ was provisioned, it's a subtle difference.

Reply:
That helped shed some light on your view, thanks!

> You can't change the _DLQ connector_ through the API or config file

i.e. a DLQ connector (configuration) is immutable?

Reply:
The DLQ configuration is mutable, I'm saying you can't change the connector entity that's created based on that configuration, it only exists internally while the pipeline is running.

Reply:
I can see your point more clearly now. Personally, I'm not fully on board with overlapping provisioning types, but I also recognize the points you made. :+1: